qpc/insights/upload.py

```diff
@@ -30,13 +30,34 @@
                                 format_upload_success)
 from qpc.request import GET, request
 from qpc.translation import _
-from qpc.utils import (validate_write_file,
+from qpc.utils import (extract_json_from_tarfile,
+                       validate_write_file,
                        write_file)
 
 # pylint:disable=no-member
 from requests import codes
 
 
+def verify_report_fingerprints(fingerprints, report_id):
+    """Verify that report fingerprints contain canonical facts."""
+    canonical_facts = ['insights_client_id', 'bios_uuid', 'ip_addresses',
+                       'mac_addresses', 'vm_uuid', 'etc_machine_id',
+                       'subscription_manager_id']
+    verified_fingerprints = []
+    for fingerprint in fingerprints:
+        found_facts = False
+        for fact in canonical_facts:
+            if fingerprint.get(fact):
+                found_facts = True
+                break
+        if found_facts:
+            verified_fingerprints.append(fingerprint)
+        else:
+            print(_(messages.INSIGHTS_REPORT_INVALID_FP %
+                    (report_id, fingerprint.pop('metadata', None))))
```

Comment:
If there is no metadata that means that a None will be returned for the fingerprint, even though there may be some other keys in there. There should always be metadata so it may not be an issue.